src/zenml/cli/secret.py

```diff
@@ -69,195 +61,292 @@ def secret(ctx: click.Context) -> None:
-            " for both."
-        )
+    try:
+        parsed_args = parse_unknown_options(args, expand_args=True)
+    except AssertionError as e:
+        error(str(e))
+        return
 
     if name == "name":
         error("Secret names cannot be named 'name'.")
 
-    secret_schema = SecretSchemaClassRegistry.get_class(
-        secret_schema=secret_schema_type
-    )
-    secret_keys = secret_schema.get_schema_keys()
+    if name.startswith("--"):
+        error(
+            "Secret names cannot start with '--' The first argument must be."
+            "the secret name."
+        )
 
-    secret_contents = {"name": name}
+    if "name" in parsed_args:
+        error("Secret names cannot be passed as arguments.")
 
-    if secret_keys:
-        click.echo(
-            "You have supplied a secret_set_schema with predefined keys. "
-            "You can fill these out sequentially now. Just press ENTER to skip "
-            "optional secrets that you do not want to set"
+    try:
+        secret_schema = SecretSchemaClassRegistry.get_class(
+            secret_schema=secret_schema_type
         )
-        for k in secret_keys:
-            v = getpass.getpass(f"Secret value for {k}:")
-            if v:
-                secret_contents[k] = v
+    except KeyError as e:
+        error(str(e))
 
-    elif secret_key and secret_value:
-        secret_contents[secret_key] = secret_value
+    secret_keys = secret_schema.get_schema_keys()
+
+    secret_contents = {"name": name}
+
+    if interactive:
+
+        if parsed_args:
+            error(
+                "Cannot pass secret fields as arguments when using interactive "
+                "mode."
+            )
+
+        if secret_schema_type != ARBITRARY_SECRET_SCHEMA_TYPE:
+            click.echo(
+                "You have supplied a secret schema with predefined keys. "
+                "You can fill these out sequentially now. Just press ENTER to "
+                "skip optional secrets that you do not want to set"
+            )
+            for k in secret_keys:
+                v = getpass.getpass(f"Secret value for {k}:")
+                if v:
+                    secret_contents[k] = v
+        else:
+            click.echo(
+                "You have not supplied a secret schema with any "
+                "predefined keys. Entering interactive mode:"
+            )
+            while True:
+                k = click.prompt("Please enter a secret-key")
+                if k not in secret_contents:
+                    secret_contents[k] = getpass.getpass(
+                        f"Please enter the secret_value " f"for the key [{k}]:"
+                    )
+                else:
+                    warning(
+                        f"Key {k} already in this secret. Please restart this"
+                        f" process or use "
+                        f"'zenml secret update {name} --{secret_schema_type}'"
```

Comment:
I think this was meant maybe?

```suggestion
                        f"'zenml secret update {name} --{key}=...'"
```

Reply:
Yes, something like that :)